src/main/java/hudson/plugins/analysis/graph/CategoryBuildResultGraph.java

```diff
@@ -234,7 +237,18 @@ protected JFreeChart createChart(final GraphConfiguration configuration, final R
         }
         return valuesPerBuild;
     }
-
+    private boolean passesFilteringByParameter(final AbstractBuild<?, ?> build, final String parameterName, final String parameterValue){
+        if (parameterName == null){
+            return true;
+        }
+        Map<String, String> vars = build.getBuildVariables();
+        if(vars.containsKey(parameterName)){
+            if (vars.get(parameterName).equals(parameterValue)) {
```

Comment:
Sorry, i saw it on different view and did not realize you pointed out the specific line. You are right there are some potential NPE's here. I will add checks.